src/graphql/object_type.cr

```diff
@@ -172,6 +82,153 @@ module GraphQL::ObjectType
       {% end %}
     end
   end
+
+  # :nodoc:
+  private def _graphql_serialize(context, field : ::GraphQL::Language::Field, value, json : ::JSON::Builder) : Array(::GraphQL::Error)
```

Comment:
Made this a plain method instead of a macro. this way we can call it recursively which makes the array case a bit simpler (despite the complexity introduced by the concurrent resolving)